server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -857,30 +858,26 @@ private static boolean failed(SnapshotInfo snapshot, String index) {
     }
 
     /**
-     * Check if any of the indices to be closed are currently being restored from a snapshot and fail closing if such an index
-     * is found as closing an index that is being restored makes the index unusable (it cannot be recovered).
+     * Returns the indices that match the given predicate and that are currently being restored.
      */
-    public static void checkIndexClosing(ClusterState currentState, Set<IndexMetaData> indices) {
-        RestoreInProgress restore = currentState.custom(RestoreInProgress.TYPE);
-        if (restore != null) {
-            Set<Index> indicesToFail = null;
-            for (RestoreInProgress.Entry entry : restore) {
-                for (ObjectObjectCursor<ShardId, RestoreInProgress.ShardRestoreStatus> shard : entry.shards()) {
-                    if (!shard.value.state().completed()) {
-                        IndexMetaData indexMetaData = currentState.metaData().index(shard.key.getIndex());
-                        if (indexMetaData != null && indices.contains(indexMetaData)) {
-                            if (indicesToFail == null) {
-                                indicesToFail = new HashSet<>();
-                            }
-                            indicesToFail.add(shard.key.getIndex());
-                        }
-                    }
+    public static Set<Index> restoringIndices(final ClusterState currentState, final Predicate<Index> predicate) {
```

Comment:
NIT: I like the move to `Index`, but the `Predicate` solution instead of just passing a collection to check seems a little strange to me since we seem to mostly use `::contains` on a collection here anyway. I'd like sticking with a collection and checking `contains` better for simplicity and debugability but it doesn't matter much to me :)

Reply:
Sure